scirisweb/sw_app.py

```diff
@@ -269,18 +304,28 @@ def run(self, with_twisted=True, with_flask=True, with_client=True, do_log=False
             for linestr in logostr.splitlines():
                 sc.colorize(logocolors, linestr, enable=self.colorize)
             print('')
-        
-        # Run the thing
-        
+
+
         if not with_twisted: # If we are not running the app with Twisted, just run the Flask app.
-            self.flask_app.run(port=port)
-        else: # Otherwise (with Twisted).
-            client_dir = self.config['CLIENT_DIR']
-            if   not with_client and not with_flask: run_twisted(port=port, do_log=do_log)  # nothing, should return error
-            if   with_client     and not with_flask: run_twisted(port=port, do_log=do_log, client_dir=client_dir)   # client page only / no Flask
-            elif not with_client and     with_flask: run_twisted(port=port, do_log=do_log, flask_app=self.flask_app)  # Flask app only, no client
-            else:                                    run_twisted(port=port, do_log=do_log, flask_app=self.flask_app, client_dir=client_dir)  # Flask + client
-        return None      
+            run_fcn = lambda: self.flask_app.run(port=port, **run_args)
+        else:
+            twisted_args = {}
+            twisted_args['reactor_args'] = run_args
+            twisted_args['do_log'] = do_log
+            twisted_args['port'] = port
+            if with_client:
+                twisted_args['client_dir'] = self.config['CLIENT_DIR']
+            if with_flask:
+                twisted_args['flask_app'] = self.flask_app
+            if autoreload:
+                twisted_args['reactor_args']['installSignalHandlers'] = 0
+            run_fcn = lambda: run_twisted(**twisted_args)
+
+        if autoreload:
+            import werkzeug.serving
```

Comment:
Any reason why this is imported here rather than at the top with the other werkzeug functions?

Reply:
No reason apart from that was in the github gist I copied in! I've moved the import now